components/style/matching.rs

```diff
@@ -977,7 +1031,10 @@ pub trait MatchMethods : TElement {
                     }
                 }
 
-                return data.set_primary_rules(rules);
+                return RulesMatchedResult {
+                    rule_nodes_changed: data.set_primary_rules(rules),
+                    important_rules_overriding_animation_changed: false,
```

Comment:
Also, I overlooked this, and this is wrong for `::before` and `::after`, isn't it? Why don't you need to know whether important rules changed here?